pagefaultgames/pokerogue
# [Bug] Disable Egg Gacha option in shop

## What are the changes?
1. This prevents the "Egg Gacha" from being opened in the Shop screen.

## Why am I doing these changes?
1. There's a Candy exploit where a user can repeatedly catch a Pokemon, open the Egg Gacha, pull an egg to have the game force save the session (with the caught Pokemon and Candy), and then reload to have the scene reset to before they caught the Pokemon (but keeping the caught Pokemon and Candy).

## What did change?
1. The exploit is no longer possible because the Egg Gacha save won't be accessible until after the session has been dealt with otherwise (e.g. exited or saved by the next wave starting).

## How to test the changes?
1. You should no longer be able to open the "Egg Gacha" while the Shop Overlay is open.


https://github.com/pagefaultgames/pokerogue/assets/6127618/7eeef858-2a8b-4922-a15f-256e5fa8296d




## Checklist
